src/go/rpk/pkg/cli/cmd/cluster/offsets.go

```diff
@@ -9,266 +9,471 @@
-	for topic, partitions := range offsets.Blocks {
-		for partition, partitionInfo := range partitions {
-			latestOffset, err := client.GetOffset(topic, partition, sarama.OffsetNewest)
-			if err != nil {
-				return nil, fmt.Errorf("Error listing consumer groups: %v", err)
-			}
-			row := ConsumerLagRow{
-				Group:           groupId,
-				Topic:           topic,
-				Partition:       partition,
-				LatestOffset:    latestOffset,
-				CommittedOffset: partitionInfo.Offset,
-			}
-			if partitions, ok := memberAssignments[topic]; ok {
-				if member, ok := partitions[partition]; ok {
-					row.MemberId = member.MemberId
-					row.ClientHost = member.ClientHost
-					row.ClientId = member.ClientId
-					member.Matched = true
-				}
-			}
-			consumers = append(consumers, row)
+	if useErr {
+		headers = append(headers, "ERROR")
+		orig := args
+		args = func(r *describeRow) []interface{} {
+			return append(orig(r), r.err)
 		}
 	}
 
-	// Also add in members that didn't have a committed offset
-	for topic := range memberAssignments {
-		for partition := range memberAssignments[topic] {
-			member := memberAssignments[topic][partition]
-			if member.Matched {
-				continue
-			}
-			latestOffset, err := client.GetOffset(topic, partition, sarama.OffsetNewest)
-			if err != nil {
-				return nil, fmt.Errorf("Error fetching partition offset: %v", err)
-			}
-			row := ConsumerLagRow{
-				Group:           groupId,
-				Topic:           topic,
-				Partition:       partition,
-				LatestOffset:    latestOffset,
-				CommittedOffset: -1,
-				MemberId:        member.MemberId,
-				ClientHost:      member.ClientHost,
-				ClientId:        member.ClientId,
+	tw = out.NewTable(headers...)
+	defer tw.Flush()
+	for _, row := range rows {
+		tw.Print(args(&row)...)
+	}
+}
+
+// Below here, we un-binary a DescribeGroupsResponse.
+//
+// Kafka actually knows nothing about what clients are assigning each other.
+// The protocol is agnostic, which makes it extremely extendable (see streams).
+// We have to take the blob of bytes and deserialize it. The standard protocol
+// is the "consumer" protocol, which deserializes into the kmsg types used
+// below. Only streams uses a different protocol (or maybe it is the same
+// protocol, but with a super advanced balancer? tough to remember).
+
+type describedGroupMember struct {
+	MemberID         string
+	InstanceID       *string
+	ClientID         string
+	ClientHost       string
+	MemberMetadata   kmsg.GroupMemberMetadata
+	MemberAssignment kmsg.GroupMemberAssignment
+}
+
+type describedGroup struct {
+	Broker               kgo.BrokerMetadata
+	ErrorCode            int16
+	Group                string
+	State                string
+	ProtocolType         string
+	Protocol             string
+	Members              []describedGroupMember
+	AuthorizedOperations int32
+}
+
+type describeGroupsResponse struct {
+	ThrottleMillis int32
+	Groups         []describedGroup
+}
+
+func unbinaryGroupDescribeMembers(
```

Comment:
unbinary = decode? unmarshal?

There must be a better word for this.

Reply:
Decode works!